d2l-focusable-behavior.html

```diff
@@ -5,6 +5,32 @@
 
 	/** @polymerBehavior */
 	D2L.PolymerBehaviors.FocusableBehavior = {
+		properties: {
+			/**
+			 * Fired when the focusable receives focus.
+			 *
+			 * @event focus
+			*/
+
+			/**
+			 * Fired when the focusable loses focus.
+			 *
+			 * @event blur
+			*/
+		},
+		attached: function() {
+			var elem = Polymer.dom(this.root).querySelector('.d2l-focusable');
+			if (!elem) return;
+			elem.addEventListener('focus', this._handleFocus.bind(this));
+			elem.addEventListener('blur', this._handleBlur.bind(this));
+		},
+
+		detached: function() {
+			var elem = Polymer.dom(this.root).querySelector('.d2l-focusable');
+			if (!elem) return;
+			elem.removeEventListener('focus', this._handleFocus.bind(this));
```

Comment:
Note that `bind` returns a new function, so the function referenced in the `attached` above is different than the function referenced here in `detached`.  The result is a no-op, and the event handler reference is not cleaned up the way we expect.

I whipped up this [little code-pen](https://codepen.io/anon/pen/bMwyyo?editors=1011) to demonstrate.

In some places you may see this sort of thing in the `ready` method to overwrite the function on the object before it's attached, so that we have a reference to the actual function that's been attached as a handler, and can properly remove it during cleanup.

```
ready: function() {
	this._someHandler = this._someHandler.bind(this);
}
```